### PR TITLE
Speed up emulator significantly.

### PR DESCRIPTION
**Pull request by sam-falvo**
This might also make MacOS version run acceptably too!

### CODE REVIEW

File: 3/src/e/e.c [hunk 1]
```diff
@@ -183,9 +184,13 @@ int run(int argc, char *argv[]) {
...
-		pp->step(p, t);
-		running &= p->running;
+		// Amortize the cost of hitting SDL sufficiently to
+		// make the emulator run at a reasonable speed.
+		for(xx=0; (xx<100000) && running; xx++) {
+			tt->pending(t);
+			pp->step(p, t);
+			running &= p->running;
+		}
 		while(SDL_PollEvent(&sdlEvent)) {
```
Comment (larsbrinkhoff) [inline 3/src/e/e.c]: @stefan, is there a lesson to be learned here?  Do we poll SDL a lot?

Comment (sam-falvo) [inline 3/src/e/e.c]: I discovered, quite by accident, that calling SDL_PollEvent is rather expensive.  Without the for-loop, text output to the MGIA frame buffer ran at a speed not unlike a TI-99/4A.  In other words, unbearably slow, especially for an 80x60 display.

By letting the CPU run for a number of cycles before polling, the speed of the emulator now more closely resembles what I'd expect from actual hardware.  E.g., video update is now about as fast as a PC/AT or Atari ST w/out a blitter.

All this time, I thought screen updates were what was taking all the time.  It turns out it's event polling!  Who would have thought?

**Comment by larsbrinkhoff** [comment]
We're calling `SDL_PollEvent` after every instruction in @stefanberndtsson's Atari ST emulator.  But it seems to run quite fine.

**Comment by sam-falvo** [comment]
SDL 1.2 or SDL 2.0?  I'm using SDL 2.0.

**Comment by sam-falvo** [comment]
In any event, the results speak for themselves; you're free to test the code with and without the for-loop and observe the difference in speed.  It's easily a factor of 10x or more improvement in performance on my Linux box.

**Comment by sam-falvo** [comment]
One bug that I cannot get rid of, again seemingly related to SDL 2.0, is that intermittently, SDL will block event processing until I move the mouse out and back inside of the emulator's window.  It doesn't always happen, indicating a race condition between threads that SDL manages, I think.  So, if you run the emulator and nothing seems to happen, try killing it and re-running it a few times.  Eventually, you'll get proper behavior.

**Comment by larsbrinkhoff** [comment]
We went from 1.2 to 2.0 a few weeks back.

You're right!  Frames per second went from 45 (already acceptable) to 75 on a decent but not top-of-the-line laptop.  Hardly a factor 10x, though.  But I'll take it!

**Comment by larsbrinkhoff** [comment]
https://github.com/stefanberndtsson/ostis if your curious and/or an Atari ST fan.

**Comment by sam-falvo** [comment]
For me, it sped my emulator up from seconds per frame to frames per second.  Though, I've never been very good at writing emulators; I think maybe my code just sucks.  :)

**Pull request merged**